react-app/src/utils/crumbs.test.ts

```diff
@@ -0,0 +1,114 @@
+// crumbs.test.ts
+import { describe, it, expect} from 'vitest';
+import { Authority } from 'shared-types/authority';
+import { Action } from 'shared-types/actions'; 
+import {
+  getDashboardTabForAuthority,
+  detailsAndActionsCrumbs,
+  dashboardCrumb,
+  detailsCrumb,
+  actionCrumb,
+} from './crumbs';
+
+
+
+describe('crumbs', () => {
+    describe('getDashboardTabForAuthority', () => {
+        it('should return "spas" for CHIP SPA', () => {
+          const result = getDashboardTabForAuthority(Authority.CHIP_SPA);
+          expect(result).toBe('spas');
+        });
+      
+        it('should return "spas" for Medicaid SPA', () => {
+          const result = getDashboardTabForAuthority(Authority.MED_SPA);
+          expect(result).toBe('spas');
+        });
+      
+        it('should return "waivers" for 1915(b)', () => {
+          const result = getDashboardTabForAuthority(Authority["1915b"]);
+          expect(result).toBe('waivers');
+        });
+      
+        it('should return "waivers" for 1915(c)', () => {
+          const result = getDashboardTabForAuthority(Authority["1915c"]);
+          expect(result).toBe('waivers');
+        });
+      
+        it('should throw an error for invalid authority', () => {
+          expect(() => getDashboardTabForAuthority("invalid" as Authority)).toThrow("Invalid authority");
+        });
+    });
+
+  describe('detailsAndActionsCrumbs', () => {
+    const id = "12345";
+    const authority = Authority.CHIP_SPA;
+
+    it('should return the default breadcrumbs without action type', () => {
+      const result = detailsAndActionsCrumbs({ id, authority });
+      expect(result).toEqual([
+        dashboardCrumb(authority),
+        detailsCrumb(id, authority),
+      ]);
```

Comment:
Do you think it'd be better to test hardcoded values as opposed to generated values? Should be harder to spoof assuming one of the functions get changed unnoticed

Reply:
generated expectedBreadcrumbs, call the following methods: action breadcrumbs, dashboard breadcrumbs, and details breadcrumbs. Doing so ensures that the test will remain accurate even if the functions change.